Objects/codeobject.c

```diff
@@ -1781,8 +1781,25 @@ code_richcompare(PyObject *self, PyObject *other, int op)
     for (int i = 0; i < Py_SIZE(co); i++) {
         _Py_CODEUNIT co_instr = _PyCode_CODE(co)[i];
         _Py_CODEUNIT cp_instr = _PyCode_CODE(cp)[i];
+
+        if (co_instr.op.code == ENTER_EXECUTOR) {
+            const int exec_index = co_instr.op.arg;
+            _PyExecutorObject *exec = co->co_executors->executors[exec_index];
+            co_instr.op.code = exec->vm_data.opcode;
```

Comment:
Comparing two code objects must not modify the code object.

The interaction between executor, the specializer and instrumentation is subtle and likely to break without care.
This will leak executors, as it flip-flops between `JUMP_BACKWARDS` and `ENTER_EXECUTOR`, or worse if an optimizer assumes that a single instruction will only be seen once.